sound/soc/sof/intel/hda-stream.c

```diff
@@ -244,6 +256,16 @@ int hda_dsp_stream_trigger(struct snd_sof_dev *sdev,
 					SOF_HDA_SD_CTL_DMA_START |
 					SOF_HDA_CL_DMA_SD_INT_MASK, 0x0);
 
+		ret = snd_sof_dsp_read_poll_timeout(sdev, HDA_DSP_HDA_BAR,
+						sd_offset, run, ((run &
+						SOF_HDA_SD_CTL_DMA_START)
+						== 0x0),
+						HDA_DSP_REG_POLL_INTERVAL_US,
```

Comment:
```c ((run & SOF_HDA_SD_CTL_DMA_START) == 0x0),``` is a lot for a parameter, it's always better to define the variable on an earlier line e.g.
```c run_value = !(run &SOF_HDA_SD_CTL_DMA_START)```
This can be fixed incrementally.

Reply:
agree with @lgirdwood please add an intermediate variable and simplify this.

Reply:
@lgirdwood @plbossart @kv2019i 
The snd_sof_dsp_read_poll_timeout is a macro #define, **not a really function**, if using predefined variable as @lgirdwood suggested, the result should be not the same. Because the #define need first get the value of "run", and then the `run & SOF_HDA_SD_CTL_DMA_START` can be calculated.
@kv2019i Are you agree?

Another solution maybe define a macro for `run & SOF_HDA_SD_CTL_DMA_START`, as:
1):
`#define RUN_VALUE  ((run) & SOF_HDA_SD_CTL_DMA_START)`

and use `RUN_VALUE == SOF_HDA_SD_CTL_DMA_START` and `RUN_VALUE == 0` as the parameter.

or
2):
`#define RUN_DMA_START  (((run) & SOF_HDA_SD_CTL_DMA_START) == SOF_HDA_SD_CTL_DMA_START)`
`#define RUN_DMA_STOP  (((run) & SOF_HDA_SD_CTL_DMA_START) == 0x0)`

and use as:
```
		ret = snd_sof_dsp_read_poll_timeout(sdev, HDA_DSP_HDA_BAR,
						sd_offset, run, RUN_DMA_START,
						HDA_DSP_REG_POLL_INTERVAL_US,
						HDA_DSP_STREAM_RUN_TIMEOUT);
```
What do you think?

Reply:
@zhuyingjiang I'm ok with the original as well -- this is used elsewhere in the codebase already. I'm not sure how much this helps readability to define more macros. Upto @plbossart and @lgirdwood who made the review comments.

One more option is to use a local variable to describe the mask only:
```
                u32 dma_start = SOF_HDA_SD_CTL_DMA_START;
		ret = snd_sof_dsp_read_poll_timeout(sdev, HDA_DSP_HDA_BAR,
						sd_offset, run, !(run & dma_start),
						HDA_DSP_REG_POLL_INTERVAL_US,
```

But yeah, the improvement is limited.

Reply:
i like this update much better...

Reply:
OK, thanks @kv2019i , I will update with this solution.